pkg/apis/sources/v1alpha2/ping_validation.go

```diff
@@ -38,6 +41,24 @@ func (cs *PingSourceSpec) Validate(ctx context.Context) *apis.FieldError {
 		errs = errs.Also(fe)
 	}
 
+	if len(cs.JsonData) > 0 && !json.Valid([]byte(cs.JsonData)) {
+		fe := apis.ErrInvalidValue("is not valid Json.", "jsonData")
+		errs = errs.Also(fe)
+	}
+
+	pingConfig := config.FromContextOrDefaults(ctx)
+	pingDefaults := pingConfig.PingDefaults.GetPingConfig()
+
+	if bsize := len(cs.JsonData); bsize > pingDefaults.JsonDataMaxBytes {
+		fe := apis.ErrInvalidValue(fmt.Sprintf("the jsonData length of %d bytes exceeds limit set at %d.", bsize, pingDefaults.JsonDataMaxBytes), "jsonData")
+		errs = errs.Also(fe)
+	}
+
+	if len(cs.JsonData) > 0 && !json.Valid([]byte(cs.JsonData)) {
```

Comment:
same comment, not backward compatible.

Reply:
We discussed in slack conversation, you agreed to json validation. Are you now rethinking that?